cns/restserver/restserver.go

```diff
@@ -1607,3 +1595,411 @@ func (service *HTTPRestService) getNumberOfCPUCores(w http.ResponseWriter, r *ht
+					for _, endpointName := range compartmentNCData {
+						if err = hnsclient.CleanupEndpoint(endpointName); err != nil {
+							returnMessage = err.Error()
+							returnCode = UnexpectedError
+							break
+						}
+					}
+
+					if err == nil {
+						if err = hnsclient.DeleteCompartment(req.CompartmentID); err != nil {
+							returnMessage = err.Error()
+							returnCode = UnexpectedError
+						} else {
+							// Compartment deleted successfully
+							returnMessage = fmt.Sprintf("Successfully deleted network compartment "+
+								"with ID: %d", req.CompartmentID)
+							service.deleteCompartmentNCData(req.CompartmentID)
+						}
+					}
+				} else {
+					returnMessage = fmt.Sprintf("ERROR: Attempting to delete invalid compartment")
+					returnCode = InvalidParameter
+				}
+			}
+		default:
+			returnMessage = fmt.Sprintf("ERROR: deleteCompartmentWithNCs API expects a DELETE")
+			returnCode = UnsupportedVerb
+		}
+	}
+
+	resp := &cns.Response{
+		ReturnCode: returnCode,
+		Message:    returnMessage,
+	}
+
+	err = service.Listener.Encode(w, &resp)
+	log.Response(service.Name, resp, resp.ReturnCode, ReturnCodeToString(resp.ReturnCode), err)
+}
+
+// Handles requests to create compartment with NCs.
+func (service *HTTPRestService) createCompartmentWithNCs(w http.ResponseWriter, r *http.Request) {
+	log.Printf("[Azure CNS] createCompartmentWithNCs")
+
+	var (
+		returnCode    int
+		returnMessage string
+		err           error
+		compartmentID int
+		req           cns.CreateCompartmentWithNCsRequest
+	)
+
+	err = service.Listener.Decode(w, r, &req)
+	log.Request(service.Name, &req, err)
+
+	if err != nil {
+		returnMessage = fmt.Sprintf("[Azure CNS] ERROR: Unable to decode input request")
+		returnCode = InvalidParameter
+	} else {
+		switch r.Method {
+		case "POST":
+			if len(req.NCIDs) <= 0 {
+				returnMessage = fmt.Sprintf("ERROR: No NCID specified in the request")
+				returnCode = InvalidParameter
+			} else if len(req.NCIDs) > hnsclient.MaxNCsPerCompartment {
+				returnMessage = fmt.Sprintf("ERROR: Number of NCIDs specified (%d) exceeds max supported (%d)",
+					len(req.NCIDs), hnsclient.MaxNCsPerCompartment)
+				returnCode = InvalidParameter
+			}
+
+			if compartmentID, err = hnsclient.CreateCompartment(); err != nil {
+				returnMessage = err.Error()
+				returnCode = UnexpectedError
+			} else {
+				networkExistsForNC := make(map[string]bool)
+				for _, ncID := range req.NCIDs {
+					var networkContainerInfo *cns.GetNetworkContainerResponse
+					if networkContainerInfo, err = service.getNetworkContainerInfo(ncID); err != nil {
+						returnMessage = fmt.Sprintf("ERROR: Cannot retrieve data for specified NCID")
+						returnCode = InvalidParameter
+					} else {
+						// check if the network for the current NC already exists.
+						if networkExistsForNC[ncID], err =
+							hnsclient.CheckNetworkExistsForNC(networkContainerInfo); err != nil {
```

Comment:
log to say network already exists for nc. Also why do we break from for loop here? Don't we need to check for remaining NCs?